tests/unit/test_charm.py

```diff
@@ -170,52 +169,6 @@ def test_saml_integration_container_down(saml_configured: Harness) -> None:
     harness.cleanup()
 
 
-def test_saml_integration_pebble_success(
-    saml_configured: Harness, monkeypatch: pytest.MonkeyPatch
-) -> None:
-    """
-    arrange: start the Synapse charm, set server_name, mock synapse.enable_saml.
-    act: call enable_saml from pebble_service.
-    assert: synapse.enable_saml is called once.
-    """
-    harness = saml_configured
-    harness.begin()
-    container = harness.model.unit.containers[synapse.SYNAPSE_CONTAINER_NAME]
-    pull_mock = MagicMock(return_value=io.StringIO("{}"))
-    push_mock = MagicMock()
-    monkeypatch.setattr(container, "pull", pull_mock)
-    monkeypatch.setattr(container, "push", push_mock)
-    enable_saml_mock = MagicMock()
-    monkeypatch.setattr(synapse, "enable_saml", enable_saml_mock)
-
-    charm_state = harness.charm.build_charm_state()
-    pebble.enable_saml(charm_state, container=container)
-
-    enable_saml_mock.assert_called_once()
-
-
-def test_saml_integration_pebble_error(
```

Comment:
Why isn't this scenario tested anymore?

Reply:
The saml observer called directly the pebble.enable_saml. Since this is not done anymore, this test was removed.

Reply:
And is there a pebble error test scenario for `enable_saml`?

Reply:
Good catch, you have a good eye for missing tests!
We don't have any to check how the charm behaves if reconcile raises a PebbleServiceError. I'll add one.

Reply:
Done: test_saml_enabled_reconcile_pebble_error, test_smtp_enabled_reconcile_pebble_error and test_redis_enabled_reconcile_pebble_error